build-ci/github_protect_travis_flake8.rst

```diff
@@ -0,0 +1,79 @@
+#############################################
+Github Branch Protection, TravisCI and Flake8
+#############################################
+
+DM recommends using `Github branch protection <https://help.github.com/articles/about-protected-branches/>`_, `TravisCI <https://travis-ci.org/>`_ and `Flake8 <http://flake8.pycqa.org/en/stable/>`_ on repositories.
+
+.. _branch-protection:
+
+Github Branch Protection
+========================
+
+All DM repositories should protect the master branch. Protected branches block some features of Git. Specifically DM protected branches can't be edited from the web, force pushed, deleted or merged until their status checks pass. Furthermore status checks must pass using the latest base branch code.
+
+.. _travisci:
+
+Travis CI
+=========
+
+TravisCI is a service that provides continuous integration for Github repositories. This service must be enabled for the repository before it will work. After it is enabled there will be a page available with the TravisCI status for the repository. For example lsst.utils `https://travis-ci.org/lsst/utils <https://travis-ci.org/lsst/utils>`_.
+
+.. _flake8:
+
+Flake8
+======
+
+`Flake8 <http://flake8.pycqa.org/en/stable/manpage.html>`_ is a command-line utility for enforcing style consistency across Python projects. It includes `PyFlakes <https://launchpad.net/pyflakes>`_ and `PEP-0008 <https://www.python.org/dev/peps/pep-0008/>`_ lint checks by default. DM uses the repository's :file:`setup.cfg` file to configure flake8. The :doc:`python style guide <../coding/python_style_guide#pep-8-is-the-baseline-coding-style>` has further details on specific lint check exceptions. The following :file:`setup.cfg` file implements DM's lint check exceptions:
+
+.. code-block:: text
+   [flake8]
+   max-line-length = 110
+   ignore = E133, E226, E228, N802, N803
+   exclude =
+       __init__.py,
+       .tests
+
+.. _sqre-gtf:
+
+sqre-gtf
+========
+
+`sqre-gtf <https://github.com/lsst-sqre/sqre-gtf>`_ provides command line interfaces to use Github branch protection, TravisCI and Flake8. sqre-gtf runs on Python 2.7, 3.5 and 3.6. You can install it with
```

Comment:
Github TravisCI and Flake8 automation. But I am not attached to the name. I just needed to name the repository.